electrum/network.py

```diff
@@ -287,7 +287,7 @@ def __init__(self, config: SimpleConfig, *, daemon: 'Daemon' = None):
         self.asyncio_loop = util.get_asyncio_loop()
         assert self.asyncio_loop.is_running(), "event loop not running"
 
-        assert isinstance(config, SimpleConfig), f"config should be a SimpleConfig instead of {type(config)}"
```

Comment:
@ecdsa I think we can safely remove this assert, daemon isn't checked either and config is dereferenced almost immediately below so that should nicely fail in the unlikely event some other object is passed in.